pkg/chunk/storage_client.go

```diff
@@ -19,8 +19,8 @@ type StorageClient interface {
 	QueryPages(ctx context.Context, query IndexQuery, callback func(result ReadBatch, lastPage bool) (shouldContinue bool)) error
 
 	// For storing and retrieving chunks.
-	PutChunk(ctx context.Context, key string, data []byte) error
-	GetChunk(ctx context.Context, key string) ([]byte, error)
+	PutChunks(ctx context.Context, chunks []Chunk, keys []string, data [][]byte) error
```

Comment:
The list must be the same length, and the ordering within them must be consistent.

You could probably consider this a micro optimization, as the chunk can generate the key and the buffer.  I'll see if I can factor it out.